gobblin-core/src/test/java/org/apache/gobblin/converter/avro/JsonElementConversionFactoryTest.java

```diff
@@ -0,0 +1,317 @@
+
+
+/**
+ * Unit test for {@link JsonElementConversionFactory}
+ *
+ * @author Tilak Patidar
+ */
+public class JsonElementConversionFactoryTest {
+
+  private static WorkUnitState state;
+  private static JsonObject testData;
+  private static JsonParser jsonParser = new JsonParser();
+
+  @Rule
+  public TestName name = new TestName();
+
+  @BeforeClass
+  public static void setUp() {
+    WorkUnit workUnit = new WorkUnit(new SourceState(),
+        new Extract(new SourceState(), Extract.TableType.SNAPSHOT_ONLY, "namespace", "dummy_table"));
+    state = new WorkUnitState(workUnit);
+    Type listType = new TypeToken<JsonObject>() {
+    }.getType();
+    Gson gson = new Gson();
+    testData = gson.fromJson(new InputStreamReader(
+            JsonElementConversionFactoryTest.class.getResourceAsStream("/converter/JsonElementConversionFactoryTest.json")),
+        listType);
+  }
+
+  @Test
+  public void schemaWithArrayOfMaps()
+      throws Exception {
+    JsonObject schema = getSchemaData().getAsJsonObject();
+    JsonObject expected = getExpectedSchema().getAsJsonObject();
+
+    ArrayConverter converter = new ArrayConverter("dummy", true, ARRAY.toString(), schema, state);
+    Assert.assertEquals(avroSchemaToJsonElement(converter), expected);
+  }
+
+  @Test
+  public void schemaWithArrayOfRecords()
+      throws Exception {
+    JsonObject schema = getSchemaData().getAsJsonObject();
+    JsonObject expected = getExpectedSchema().getAsJsonObject();
+
+    ArrayConverter converter = new ArrayConverter("dummy1", true, ARRAY.toString(), schema, state);
+
+    Assert.assertEquals(avroSchemaToJsonElement(converter), expected);
+  }
+
+  @Test
+  public void schemaWithRecord()
+      throws DataConversionException, SchemaConversionException, UnsupportedDateTypeException {
+    JsonObject schema = getSchemaData().getAsJsonObject();
+    JsonObject expected = getExpectedSchema().getAsJsonObject();
+
+    RecordConverter converter = new RecordConverter("dummy1", true, RECORD.toString(), schema, state,
+        buildNamespace(state.getExtract().getNamespace(), "something"));
+
+    Assert.assertEquals(avroSchemaToJsonElement(converter), expected);
+  }
+
+  @Test
+  public void schemaWithArrayOfInts()
+      throws Exception {
+    JsonObject schema = getSchemaData().getAsJsonObject();
+    JsonObject expected = getExpectedSchema().getAsJsonObject();
+
+    ArrayConverter converter = new ArrayConverter("dummy1", true, ARRAY.toString(), schema, state);
+
+    Assert.assertEquals(avroSchemaToJsonElement(converter), expected);
+  }
+
+  @Test
+  public void schemaWithNullType() {
+    NullConverter nullConverter = new NullConverter("dummy1", ARRAY.toString());
+    JsonObject expected = new JsonObject();
+    expected.addProperty("type", "null");
+    expected.addProperty("source.type", "array");
+
+    Assert.assertEquals(avroSchemaToJsonElement(nullConverter), expected);
+  }
+
+  @Test
+  public void schemaWithArrayOfEnums()
```

Comment:
This test and `schemaWithMapOfEnum` fails for me.

Reply:
@jack-moseley tests are passing for me. Were the resources loaded when tests ran? Could you share the failing message?

Reply:
@jack-moseley found the failing test